pkg/cmd/server/plugin/plugin.go

```diff
@@ -30,65 +30,54 @@ import (
 )
 
 func NewCommand() *cobra.Command {
+	objectStores := map[string]cloudprovider.ObjectStore{
+		"aws":   aws.NewObjectStore(),
+		"gcp":   gcp.NewObjectStore(),
+		"azure": azure.NewObjectStore(),
+	}
+
+	blockStores := map[string]cloudprovider.BlockStore{
+		"aws":   aws.NewBlockStore(),
+		"gcp":   gcp.NewBlockStore(),
+		"azure": azure.NewBlockStore(),
+	}
+
 	c := &cobra.Command{
 		Use:    "plugin [TYPE] [NAME]",
 		Hidden: true,
 		Short:  "INTERNAL COMMAND ONLY - not intended to be run directly by users",
 		Run: func(c *cobra.Command, args []string) {
 			if len(args) != 2 {
-				log.Fatalf("You must specify exactly two arguments, the plugin type and the plugin name")
+				log.Fatalf("You must specify exactly two arguments, the plugin kind and the plugin name")
 			}
 
-			pluginType := args[0]
-			pluginName := args[1]
-
-			log.Printf("Running plugin command for type=%s, name=%s", pluginType, pluginName)
+			kind := args[0]
+			name := args[1]
 
-			switch pluginType {
-			case "objectstore":
-				var objectStore cloudprovider.ObjectStore
+			log.Printf("Running plugin command for kind=%s, name=%s", kind, name)
 
-				switch pluginName {
-				case "aws":
-					objectStore = aws.NewObjectStore()
-				case "gcp":
-					objectStore = gcp.NewObjectStore()
-				case "azure":
-					objectStore = azure.NewObjectStore()
-				default:
-					log.Fatalf("Unrecognized plugin name %q", pluginName)
+			switch kind {
+			case "cloudprovider":
```

Comment:
This makes it harder to support overriding (see comment below).

Reply:
I changed how discovery works in the manager, but I'm not sure I understand this comment - could you clarify?